### ISSUE TITLE
Diacritics row

### ISSUE THREAD

**Issue by Athozus** [maliit/keyboard#234]
Hi,

First, thanks for the amazing keyboard you're providing, it's really functional. But, I'm French, so native French speaker. And French is using A LOT of accents.

I know we can type accents by long pressing, but when you're writing a small phrase with a lot of accents, that's frustating. So I was wondering if we could implement a row (like the numbers row we can see on some other mobiles) displaying diacritics, such as accents.

This would be a row of radio button (plus you can select none of it), and that would replace letters by their diacritized version, or hide them if it doesn't exist.

Thanks !

**Comment by PureTryOut** [comment]
What does the keyboard on Android do? Can we not provide a layout for the French language that has those keys better accessible? The same would probably apply to a lot of languages.
I know in Norwegian for example they have characters like ø and å, and on the Norwegian keyboard layout they just have those keys directly available. I'd prefer doing that than trying to make some generic solution that would then really only work for a few languages.

**Comment by Athozus** [comment]
> What does the keyboard on Android do?

Check that post : https://www.reddit.com/r/gboard/comments/18z2jfd/need_help_with_number_row_in_gboard/

> Can we not provide a layout for the French language that has those keys better accessible? The same would probably apply to a lot of languages. I know in Norwegian for example they have characters like ø and å, and on the Norwegian keyboard layout they just have those keys directly available.

I think that we could just have different lists of diacritics, like we have different layouts of keys.

> I'd prefer doing that than trying to make some generic solution that would then really only work for a few languages.

A toggle may be better. In a "classic" message in French, you will have at least one but probably several of those : `éèêëàâäîïôöùûüŷÿç`, and probably skipping some (really most common are `éèà`). That's too narrow to put those diacritics accessibles.

**Comment by dobey** [comment]
> What does the keyboard on Android do?

Depends on which one, but the default is generally the same for all, where some latin languages may have an extra key or two (such as ñ for Spanish/Catalan), and relies more on either using long-press to type accents directly, or on autocomplete matching to select appropriate spelling in the given language. I don't think adding separate buttons to the OSK for accents is the solution for whatever troubles one might have typing in other languages, here.

I would be more interested in understanding the problems exactly, than a request for a feature that makes the keyboard more complex to use, without actually improving the ability to use the keyboard.

**Comment by Athozus** [comment]
> > What does the keyboard on Android do?
> 
> Depends on which one, but the default is generally the same for all, where some latin languages may have an extra key or two (such as ñ for Spanish/Catalan), and relies more on either using long-press to type accents directly, or on autocomplete matching to select appropriate spelling in the given language. I don't think adding separate buttons to the OSK for accents is the solution for whatever troubles one might have typing in other languages, here.

As I said before, I'd like the row I propose to be togglable. The diacritics keys on it would be like your trema key that allow you to write a trema over letters.

> I would be more interested in understanding the problems exactly, than a request for a feature that makes the keyboard more complex to use, without actually improving the ability to use the keyboard.

The problem is for languages such as French which uses a lot of accents, waiting 1 second for each accent is kind of frustrating.

**Comment by dobey** [comment]
> The problem is for languages such as French which uses a lot of accents, waiting 1 second for each accent is kind of frustrating.

The extended press time is I think around 300ms, not a full second. Tapping a key to show a specific accent on characters that have it, waiting for things to redraw, and then typing the character, is going to take longer. Also, do you not have `hunspell-fr` or whatever the appropriate package is for the spelling dictionary, installed? Or do you have spell checking and autocorrect disabled?

The real solution to the problems you describe here, isn't more keys on the keyboard, but improving completion, and I would say it's the same for Chinese with Pinyin input.

One particular problem I've noticed, not with maliit itself, but with packaging of the keyboard, on most distros, is that the language plugins in maliit are not split into separate packages, and there are no dependencies on the language dictionaries for hunspell, so people using Plasma Mobile are just not getting any of the benefits of the spell checker.

**Comment by Athozus** [comment]
> The extended press time is I think around 300ms, not a full second.

Yes, of course it's not a second, but plus the reaction time it's > 500ms. 

> Tapping a key to show a specific accent on characters that have it, waiting for things to redraw, and then typing the character, is going to take longer.

I'm not sure. For example, when you want to write a number, you have the time to go in secondary keyboard, press the number, then go back to the primary keyboard, and the total is shorter than writing a single accent (I can confirm you, as I said, French is using a lot of accents). This would be the same of the extra diacritics row, because the transition would be immediate, so it would become "mechanic".

> Also, do you not have hunspell-fr or whatever the appropriate package is for the spelling dictionary, installed? Or do you have spell checking and autocorrect disabled?

I've disabled it because it smashed all my words once I pressed backspace (and it still does, but lesser).

> The real solution to the problems you describe here, isn't more keys on the keyboard, but improving completion, and I would say it's the same for Chinese with Pinyin input.

Even on "large" dictionaries used in apps such as Firefox, LibreOffice, or also the Android dictionary, often words aren't detected because in our language we're using a lot of agreements (not sure about the translation of this word). Having dictionaries is great, but I can ensure you that it will not fix the problem (and maybe not all accents variations of a single word will not be displayed on a single line).

> One particular problem I've noticed, not with maliit itself, but with packaging of the keyboard, on most distros, is that the language plugins in maliit are not split into separate packages, and there are no dependencies on the language dictionaries for hunspell, so people using Plasma Mobile are just not getting any of the benefits of the spell checker.

Iirw, spell checker doesn't work. But for the moment, that's not what I'm looking for I think.

=> Of course, I say all of that but understand that it is not your priority, because languages such as mine are not in majority.
